File: 2049-eliminate-maximum-number-of-monsters/eliminate-maximum-number-of-monsters.js
/**
 * @param {number[]} dist
 * @param {number[]} speed
 * @return {number}
 */
var eliminateMaximum = function(dist, speed) {
    let time = [];
    let n = dist.length;
    for(let i=0; i<n; i++) {
        time.push(dist[i]/speed[i]);
    }
    time.sort((a,b) => a-b);
    for(let i=0; i<n; i++) {
        if(i >= time[i]) return i;
    }
    return n;
};

File: 2049-eliminate-maximum-number-of-monsters/eliminate-maximum-number-of-monsters.java
class Solution {
    public int eliminateMaximum(int[] dist, int[] speed) {
        int n = dist.length;
        int[] time = new int[n];
        for(int i=0; i<n; i++) {
            time[i] = (int)Math.ceil(dist[i]/(double)speed[i]);
        }
        Arrays.sort(time);
        //System.out.println(Arrays.toString(time));
        int count = 0;
        for(int i=0; i<n; i++) {
            if(count < time[i]) count++;
            else return count;
        }
        return count;
    }
}